app/controllers/application_controller.rb

```diff
@@ -48,6 +48,20 @@ def default_section_path(educator)
     no_sections_path
   end
 
+  # Wrap all database queries with this to enforce authorization
+  def authorized(&block)
+    authorizer.authorized(&block)
+  end
+
+  # Enforce authorization and raise if no authorized models
+  def authorized_or_raise!(&block)
```

Comment:
I can't see anywhere in the code that calls this method.

Reply:
@alexsoble Yep, it's not used here.  The idea is that it's for controller code doing lookups like:

```
def show
  student = authorized_or_raise! { Student.find(params[:id]) }
  ...
```
 vs.
```
def index
  students = authorized { Student.all }
  ...
```

We could get rid of it if we wanted to simplify, but maybe worth sharing more about why it's here and some other pieces of next steps.

So after we talked about this `authorized` API I made it and then tried to use it with the `students_controller`.  That work is over here: https://github.com/studentinsights/studentinsights/pull/1237/files#diff-11751d644b56a1cce6d8dfc73e331b91R21 and here is a smaller example that shows better what's involved: https://github.com/studentinsights/studentinsights/commit/febdb0883399814a94d64a217efc5575eb640d13.

That involved several other changes: either refactoring the controller code, updating the test setup, or adding other models like Section to the authorizer code.  It also seemed like it'd be good to revise the `students_controller_spec` to use the same test setup code in `TestPals` that is used in the `authorizer_spec` here.  All that seemed like it was getting to be a big change, so I pulled off this piece of work to merge it first.  So that's why I left this in here, and probably should have explained more.

Reply:
OK that makes a lot of sense! So basically `authorized_or_raise!` would be used for an individual record like a Student or an EventNote.